doc/api/url.md

```diff
@@ -1029,6 +1029,50 @@ new URL('/some/path%.c', 'file:'); // Incorrect: file:///some/path%.c
 pathToFileURL('/some/path%.c');    // Correct:   file:///some/path%25.c (POSIX)
 ```
 
+### `url.urlToHttpOptions(url)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `url` {URL} The [WHATWG URL][] object to convert to an options object.
+* Returns: {Object} Options object
+  * `protocol` {string} Protocol to use.
+  * `hostname` {string} A domain name or IP address of the server to issue the
+    request to.
+  * `hash` {string} The fragment portion of the URL.
+  * `search` {string} The serialized query portion of the URL.
+  * `pathname` {string} The path portion of the URL.
+  * `path` {string} Request path. Should include query string if any.
+    E.G. `'/index.html?page=12'`. An exception is thrown when the request path
+    contains illegal characters. Currently, only spaces are rejected but that
+    may change in the future.
+  * `href` {string} The serialized URL.
+  * `port` {number} Port of remote server.
+  * `auth` {string} Basic authentication i.e. `'user:password'` to compute an
+    Authorization header.
+
+This utility function converts a URL object into an ordinary options object as
+expected by the [`http.request()`][] and [`https.request()`][] APIs.
+
+```js
+const { urlToHttpOptions } = require('url');
+const myURL = new URL('https://a:b@測試?abc#foo');
```

Comment:
Sorry for intruding here - but just out of curiosity: what's `urlToHttpOptions` doing that you wouldn't get out of a `URL` already, exactly? Looks almost 1:1 to me.


```js
> new URL('https://a:b@測試?abc#foo')
URL {
  href: 'https://a:b@xn--g6w251d/?abc#foo',
  origin: 'https://xn--g6w251d',
  protocol: 'https:',
  username: 'a',
  password: 'b',
  host: 'xn--g6w251d',
  hostname: 'xn--g6w251d',
  port: '',
  pathname: '/',
  search: '?abc',
  searchParams: URLSearchParams { 'abc' => '' },
  hash: '#foo'
}
```

Reply:
`URL` does not have an `auth` option, `port` is a string (it should be a number or undefined), it doesn't have a `path` property. These options are used by the `http` module.

Reply:
Thanks for the clarification 👍🏼. Exactly - that's why I was saying "almost 1:1". I mean,


```js
function urlToHttpOptions(url) {
  return {
    ...url,
    auth: `${url.username}:${url.password}`.
    port: url.port === '' ? undefined : Number(url.port),
    path: `${url.pathname ?? ''}${url.search ?? ''}`
  }
}
```

Not 100%, but that's pretty much it, right?